code/game/machinery/flatpacker.dm

```diff
@@ -0,0 +1,296 @@
+	max_part_tier = servo.tier
+	flatpack_time = initial(flatpack_time) - servo.tier/2 // T4 = 2 seconds off
+	var/efficiency = initial(creation_efficiency)
+	for(var/datum/stock_part/micro_laser/laser in component_parts)
+		efficiency -= laser.tier * 0.2
+	creation_efficiency = max(1.2,efficiency)
+
+/obj/machinery/flatpacker/examine(mob/user)
+	. += ..()
+	if(in_range(user, src) || isobserver(user))
+		. += span_notice("The status display reads:")
+		. += span_notice("Capable of packing up to <b>Tier [max_part_tier]</b>.")
+		. += span_notice("Storing up to <b>[materials.max_amount]</b> material units.")
+		. += span_notice("Material consumption at <b>[creation_efficiency*100]%</b>")
+
+/obj/machinery/flatpacker/proc/AfterMaterialInsert(container, obj/item/item_inserted, last_inserted_id, mats_consumed, amount_inserted, atom/context)
+	SIGNAL_HANDLER
+
+	flick_overlay_view("[base_icon_state]_[item_inserted.has_material_type(/datum/material/glass) ? "glass" : "metal"]", 1.4 SECONDS)
+
+	directly_use_energy(min(active_power_usage * 0.25, amount_inserted / 100))
+
+	update_static_data_for_all_viewers()
+
+/obj/machinery/flatpacker/update_overlays()
+	. = ..()
+
+	if(!isnull(inserted_board))
+		. += mutable_appearance(icon, "[base_icon_state]_c")
+
+/obj/machinery/flatpacker/ui_interact(mob/user, datum/tgui/ui)
+	ui = SStgui.try_update_ui(user, src, ui)
+
+	if(!ui)
+		ui = new(user, src, "Flatpacker")
+		ui.open()
+
+/obj/machinery/flatpacker/ui_static_data(mob/user)
+	var/list/data = materials.ui_static_data()
+
+	data["SHEET_MATERIAL_AMOUNT"] = SHEET_MATERIAL_AMOUNT
+
+	return data
+
+/obj/machinery/flatpacker/ui_data(mob/user)
+	var/list/data = list()
+
+	var/atom/build = initial(inserted_board.build_path)
+	data["materials"] = materials.ui_data()
+	data["boardInserted"] = !isnull(inserted_board)
+	data["busy"] = busy
+	var/list/cost_mats = list()
+	for(var/datum/material/mat_type as anything in needed_mats)
+		var/list/new_entry = list()
+		new_entry["name"] = initial(mat_type.name)
+		new_entry["count"] = needed_mats[mat_type]
+		cost_mats += list(new_entry)
+
+	var/list/design
+	if(data["boardInserted"])
+		var/disableReason = ""
+		if(!materials.has_materials(needed_mats))
+			disableReason += "Not enough materials. "
+		if(print_tier > max_part_tier)
+			disableReason += "This design is too advanced for this machine. "
+		design = list(
+			"name" = initial(build.name),
+			"requiredMaterials" = cost_mats,
+			"icon" = icon2base64(icon(initial(build.icon), initial(build.icon_state), frame = 1)),
+			"canPrint" = materials.has_materials(needed_mats) && print_tier <= max_part_tier,
+			"disableReason" = disableReason
+		)
+	data["design"] = design
+	return data
+
+/obj/machinery/flatpacker/ui_assets(mob/user)
+	return list(
+		get_asset_datum(/datum/asset/spritesheet/sheetmaterials),
+		get_asset_datum(/datum/asset/spritesheet/research_designs),
+	)
+
+/obj/machinery/flatpacker/attackby(obj/item/attacking_item, mob/living/user, params)
```

Comment:
replace `attackby()` with `item_interaction_with()`, along with the new return values `ITEM_INTERACT_SUCCESS` or `ITEM_INTERACT_BLOCKING`. you can also return `NONE` if you don't want to handle that item interaction, see the docs